ansible/plugins/callback/datadog_callback.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/python
+# Modified By : Nitigya Sharma
+# Forked from : https://github.com/DataDog/ansible-datadog-callback
+#               /blob/beb95e7dbfc13c2c793daf7de05824ed0923b961/datadog_callback.py
+# Changes :
+    # 1) Disable in check mode
+    # 2) Read API_KEY from vault variables
+    # 3) Works for pushing metrics to correct host
+
+
+
+import getpass
+import logging
+import os
+import time
+import yaml
+
+try:
+    import datadog
+    import yaml
+    HAS_MODULES = True
+except ImportError:
+    HAS_MODULES = False
+
+
+from ansible.plugins.callback import CallbackBase
+from __main__ import cli
+
+
+class CallbackModule(CallbackBase):
+    def __init__(self):
+        if not HAS_MODULES:
+            self.disabled = True
+            print('Datadog callback disabled.\nMake sure you call all required libraries: "datadog" and "yaml".')
+        elif cli and cli.options.check:
+            self.disabled = True
+            print ('Datadog callback disabled in "check mode".  ')
+        else:
+            self.disabled = False
+            # Set logger level - datadog api and urllib3
+            for log_name in ['requests.packages.urllib3', 'datadog.api']:
+                self._set_logger_level(log_name)
+
+        self._playbook_name = None
+        self._start_time = time.time()
+        self._options = None
+        if cli:
+            self._options = cli.options
+
+        # self.playbook is set in the `v2_playbook_on_start` callback method
+        self.playbook = None
+        # self.play is set in the `playbook_on_play_start` callback method
+        self.play = None
+
+    # Set logger level
+    def _set_logger_level(self, name, level=logging.WARNING):
+        try:
+            log = logging.getLogger(name)
+            log.setLevel(level)
+            log.propagate = False
+        except Exception as e:
+            # We don't want Ansible to fail on an API error
+            print("Couldn't get logger - %s" % name)
+            print(e)
+
+    # Set host to match datadog agent
+
+    def _get_hostname(self, host):
+        return self.hostvars[str(host)]['hostname']
+
+    def _generate_datadog_hostname(self, host=None):
+        try:
+            if host is not None:
+                hostname = self._get_hostname(host)
+                host = str(hostname) + "." +self.variables['internal_domain_name']
+        except Exception as e:
```

Comment:
@NitigyaS this comment still applies. Catching `Exception` is a bit of an antipattern since it can mask exceptions that you might really care about. As @dannyroberts says it would be better to catch the actual exception that could get raised.